File: bindings/java/src/org/raml/amf/core/document/EncodesDomainModel.java
package org.raml.amf.core.document;

import org.raml.amf.core.domain.DomainModel;
import org.raml.amf.core.exceptions.InvalidModelException;

/**
 * Created by antoniogarrote on 04/05/2017.
 */
public interface EncodesDomainModel {

    /**
     * Encoded domain element described in the document element.
     * @return DomainElement encoded in the document.
     * @throws InvalidModelException
     */
    public DomainModel encodes() throws InvalidModelException;

}

File: bindings/js/src/core/document/EncodesDomainModel.ts
/**
 * Created by antoniogarrote on 05/05/2017.
 */

import {DomainModel} from "../domain/DomainModel";

export interface EncodesDomainModel {
    /**
     * Encoded domain element described in the document element.
     * @return DomainElement encoded in the document.
     * @throws InvalidModelException
     */
    encodes(): DomainModel;
}